src/editor/ui_menus/main_menu_load_or_save_map.cc

```diff
@@ -67,56 +71,45 @@ MainMenuLoadOrSaveMap::MainMenuLoadOrSaveMap(EditorInteractive& parent,
      cancel_(&button_box_, "cancel", 0, 0, 0, 0, UI::ButtonStyle::kWuiSecondary, _("Cancel")),
 
      // Options
-     basedir_(basedir),
-     has_translated_mapname_(false),
-     showing_mapnames_(false) {
+     basedir_(basedir) {
 
 	g_fs->ensure_directory_exists(basedir_);
 	curdir_ = basedir_;
 
-	main_box_.add(&show_mapnames_box_, UI::Box::Resizing::kFullSize);
 	main_box_.add(&table_and_details_box_, UI::Box::Resizing::kExpandBoth);
 	main_box_.add_space(padding_);
 	main_box_.add(&table_footer_box_, UI::Box::Resizing::kFullSize);
 	main_box_.add(&directory_info_, UI::Box::Resizing::kFullSize);
 	main_box_.add_space(padding_);
 	main_box_.add(&button_box_, UI::Box::Resizing::kFullSize);
 
-	show_mapnames_box_.add(&show_mapnames_);
-	cb_dont_localize_mapnames_ =
-	   /** TRANSLATORS: Checkbox title. If this checkbox is enabled, map names aren't translated. */
-	   new UI::Checkbox(&show_mapnames_box_, Vector2i::zero(), _("Show original map names"));
-	cb_dont_localize_mapnames_->set_state(false);
-	show_mapnames_box_.add_space(2 * padding_);
-	show_mapnames_box_.add(cb_dont_localize_mapnames_, UI::Box::Resizing::kFullSize);
-	show_mapnames_box_.add_inf_space();
-
 	table_.set_column_compare(0, [this](uint32_t a, uint32_t b) { return compare_players(a, b); });
 	table_.set_column_compare(1, [this](uint32_t a, uint32_t b) { return compare_mapnames(a, b); });
 	table_.set_column_compare(2, [this](uint32_t a, uint32_t b) { return compare_size(a, b); });
 
-	table_and_details_box_.add(&table_, UI::Box::Resizing::kExpandBoth);
+	table_box_.add(&display_mode_, UI::Box::Resizing::kFullSize);
+	table_box_.add(&table_, UI::Box::Resizing::kExpandBoth);
+	table_and_details_box_.add(&table_box_, UI::Box::Resizing::kExpandBoth);
 	table_and_details_box_.add_space(0);
 	table_and_details_box_.add(&map_details_box_, UI::Box::Resizing::kFullSize);
 	map_details_box_.add(&map_details_, UI::Box::Resizing::kExpandBoth);
 
+	display_mode_.add(_("File names"), MapData::DisplayType::kFilenames);
+	display_mode_.add(_("Original map names"), MapData::DisplayType::kMapnames);
+	display_mode_.add(
+	   _("Localized map names"), MapData::DisplayType::kMapnamesLocalized, nullptr, true);
```

Comment:
```suggestion
	   _("Translated map names"), MapData::DisplayType::kMapnamesLocalized, nullptr, true);
```
Normal people don't know what "localized" means ;)

Also, we should only add this option when the current locale != "en".

Reply:
How can I detect the language? If I set it to English in the options menu, `i18n::get_locale()` will return `en_US.utf-8`. If I start Widelands with `LANG=xyz` the game is also in English and `get_locale()` returns `xyz`.

Reply:
There's a variable `has_translated_mapname_` that is being used for that on master.

Reply:
I had to remove that one: we need to create the dropdown entries once, before filling in the table, but `has_translated_mapname_` was set only when the map files were being preloaded in `fill_table()`. So we can no longer use this.

Reply:
How about something like this:

`if (i18n::get_locale() != "en" && !boost::startswith("en_", i18n::get_locale())`